### PR TITLE
COMP-580 malformed tokens file

### PR DESCRIPTION
**Pull request by q-mat-beu**
* abort logout on invalid tokens file
* normal login on invalid tokens file
* report invalid tokens file in status
* use validated token and config classes

### CODE REVIEW

File: src/cortex_cli/cortex_cli.py [hunk 1]
```diff
@@ -410,44 +443,38 @@ def login(  # pylint: disable=too-many-arguments, too-many-locals, too-many-bran
...
-                logger.info(
-                    'Existing token was used to refresh the auth session. Token manager started in foreground...'
-                )
-                start_token_manager(refresh_period, config)
-            else:
-                logger.info('Existing token was used to refresh the auth session. Token manager daemon started.')
-                daemonize_token_manager(refresh_period, config)
+            refresh_token = _validate_tokens_file(tokens_file).refresh_token
+        except (click.FileError, ValidationError):
+            click.echo('Provided tokens.json file is invalid, continuing with login wit username and password.')
```
Comment (ollityrkko) [inline src/cortex_cli/cortex_cli.py:458]: typo: wit -> with

File: src/cortex_cli/cortex_cli.py [hunk 2]
```diff
@@ -378,6 +362,55 @@ def _validate_cortex_cli_auth_login(no_daemon, no_refresh, config_file) -> dict:
     return config
 
 
+def _refresh_tokens(  # pylint: disable=too-many-arguments
+    refresh_period: int,
+    no_daemon: bool,
+    no_refresh: bool,
+    config: ConfigFile,
```
Comment (ollityrkko) [inline src/cortex_cli/cortex_cli.py:369]: `auth_server_url`, `realm` and `client_id` are stored in `config`, why are they specified here also separately?

File: src/cortex_cli/cortex_cli.py [hunk 3]
```diff
@@ -378,6 +362,55 @@ def _validate_cortex_cli_auth_login(no_daemon, no_refresh, config_file) -> dict:
...
+    auth_server_url: str,
+    realm: str,
+    client_id: str,
+    tokens_file: str,
+    refresh_token: str,
```
Comment (ollityrkko) [inline src/cortex_cli/cortex_cli.py:374]: If tokens file exists and is valid `refresh_token` can (should?) be read from there. Why is it specified here separately?

**Review by ollityrkko — COMMENTED** [review]
Interface of `_refresh_tokens` could be simpler.

File: src/cortex_cli/cortex_cli.py (hunk 2 above)
Comment (q-mat-beu) [reply src/cortex_cli/cortex_cli.py:369]: Good point, I relied on "extract method" 😅 , will fix it.

File: src/cortex_cli/cortex_cli.py (hunk 3 above)
Comment (q-mat-beu) [reply src/cortex_cli/cortex_cli.py]: Changed accordingly

File: src/cortex_cli/cortex_cli.py (hunk 2 above)
Comment (ollityrkko) [reply src/cortex_cli/cortex_cli.py:369]: It probably doesn't need the `disable=too-many-arguments` directive any more.

**Review by ollityrkko — DISMISSED** [review]

File: src/cortex_cli/cortex_cli.py [hunk 4]
```diff
@@ -274,6 +255,8 @@ def init(  # pylint: disable=too-many-arguments
         if pid:
             logger.info('Active token manager (PID %s) will be killed.', pid)
             Process(pid).terminate()
+        # Remove tokens file to start from scratch after init
+        os.remove(tokens_file)
```
Comment (freetonik) [inline src/cortex_cli/cortex_cli.py:259]: If file exists, it is removed. Also, if file exists, the that means the user confirmed they want to overwrite it (in the `_validate_path` callback). I see how these two things are in sync, but have a feeling it's a bit too loose. At first I thought the file should be deleted inside of `_validate_path`, but that's semantically wrong. So, I guess, I'm just thinking out loud, and nothing should be changed :)

**Review by freetonik — APPROVED** [review]

**Pull request merged**